server/pulp/server/async/tasks.py

```diff
@@ -255,7 +255,7 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         # To avoid the race condition where __call__ method below is called before
         # this change is propagated to all db nodes, using an 'upsert' here and setting
         # the task state to 'waiting' only on an insert.
-        task_status.save(fields_to_set_on_insert=['state', 'start_time'])
+        task_status.save(fields_to_set_on_insert=['state', 'start_time'], safe=True)
```

Comment:
@bmbouter I believe we had been using the deprecated safe=True before. Though deprecated, I believe it was safer than w=1 would be, and guaranteed that the write hit more than one node. In other words, I think not calculating the write concern here and doing just the fsync will be a regression from what we had before.

I agree that we should perform the work described in issue number 1, but IMO we should always write our code with Mongo replica sets in mind regardless of whether we have or have not tested that feature yet. Since we do use safe now, we must calculate the appropriate write concern as we replace it or we will lose ground on our HA story.